presto-hive-common/src/main/java/com/facebook/presto/hive/HdfsContext.java

```diff
@@ -40,28 +40,29 @@ public HdfsContext(ConnectorIdentity identity)
         this.clientInfo = Optional.empty();
     }
 
+    public HdfsContext(ConnectorSession session)
+    {
+        this(session, Optional.empty(), Optional.empty());
+    }
+
     public HdfsContext(ConnectorSession session, String schemaName)
     {
-        requireNonNull(session, "session is null");
-        requireNonNull(schemaName, "schemaName is null");
-        this.identity = requireNonNull(session.getIdentity(), "session.getIdentity() is null");
-        this.source = requireNonNull(session.getSource(), "session.getSource()");
-        this.queryId = Optional.of(session.getQueryId());
-        this.schemaName = Optional.of(schemaName);
-        this.tableName = Optional.empty();
-        this.clientInfo = session.getClientInfo();
+        this(session, Optional.of(schemaName), Optional.empty());
     }
 
     public HdfsContext(ConnectorSession session, String schemaName, String tableName)
+    {
+        this(session, Optional.of(schemaName), Optional.of(tableName));
```

Comment:
requireNonNull for schemaName and tableName

Reply:
Wouldn't requireNonNull already in the last constructor be enough?

`this.schemaName = requireNonNull(schemaName, "schemaName is null");`
`this.tableName = requireNonNull(tableName, "tableName is null");`

Reply:
They are completely different. This is of type `String`; the other is of type `Optional<String>`. A null input here will throw NPE.

Reply:
Understood, thanks.